### PR TITLE
Recipe updates and introductions in Tuntap family

### PR DESCRIPTION
**Pull request by jimklimov**
Revised tuntap component recipe (same version and build rules).

Added bridge module, tunctl and tuntap-docs (copy of Kazuyoshi's web page on the projects).

There are no interdependencies, so these may be merged together and built in parallel ;)

### CODE REVIEW

File: components/network/tuntap/Makefile [hunk 1]
```diff
@@ -48,10 +51,7 @@ CONFIGURE_OPTIONS.64 +=  --enable-64bit
...
 # common targets
-build:          $(BUILD_32_and_64)
+build:		$(BUILD_32_and_64)
 
 install:	$(INSTALL_32_and_64)
-
-BUILD_PKG_DEPENDENCIES =	$(BUILD_TOOLS)
-
-include $(WS_TOP)/make-rules/depend.mk
+# Auto-generated contents below.  Please manually verify and remove this comment
```
Comment (pyhalov) [inline components/network/tuntap/Makefile]: Remove this comment...

File: components/network/tuntap/tuntap-docs.p5m [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+# 
+# Copyright 2013, Adam Stevko. All rights reserved.
```
Comment (pyhalov) [inline components/network/tuntap/tuntap-docs.p5m]: Correct copyright

File: components/network/bridge/Makefile [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+COMPONENT_FMRI=		driver/network/brdg
+COMPONENT_SRC=		kaizawa-$(COMPONENT_NAME)-$(COMPONENT_GIT_REV)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:4eb9d76152081284d7f1ff5c7a60116d463bffc1a0724f5ecda85592db09b283
+COMPONENT_PROJECT_URL=	http://www.whiteboard.ne.jp/~admin2/tuntap/
+COMPONENT_ARCHIVE_URL= 	https://codeload.github.com/kaizawa/$(COMPONENT_NAME)/legacy.tar.gz/$(COMPONENT_GIT_REV)
+COMPONENT_CLASSIFICATION=	Drivers/Networking
+COMPONENT_SUMMARY=	The 'brdg' is a STREAMS module which enables briging 2 interfaces
+COMPONENT_DESTRIPTION=	$(COMPONENT_SUMMARY), \
+and 'brdgadm' is a configuration command to add/delete bridge interfaces. \
```
Comment (pyhalov) [inline components/network/bridge/Makefile:31]: 1) misprint in desTription, 2) Are you sure this value is correctly escaped (please, check with pkg info)?

**Comment by pyhalov** [comment]
@xenol: I'd appreciate your review of this work

**Comment by jimklimov** [comment]
Thanks for the many good catches, should be fixed now.

File: components/network/tunctl/Makefile [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+COMPONENT_PREP_ACTION = \
+        (cd $(@D) && rm -f if_tun.h && autoreconf -fiv)
+REQUIRED_PACKAGES += driver/network/header-tun
+
+PATCH_LEVEL = 0
+
+CFLAGS +=	$(CPP_LARGEFILES)
+LDFLAGS +=	$(LD_Z_DEFS) $(LD_Z_TEXT) -lpthread
+
+PKGLINT=true
```
Comment (pyhalov) [inline components/network/tunctl/Makefile]: What's wrong with pkglint?

File: components/network/bridge/Makefile [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+PATCH_LEVEL = 0
+
+CFLAGS +=	$(CPP_LARGEFILES)
+LDFLAGS +=	$(LD_Z_DEFS) $(LD_Z_TEXT) -lpthread
+
+CONFIGURE_OPTIONS.32 +=  --disable-64bit
+CONFIGURE_OPTIONS.64 +=  --enable-64bit
+
+PKGLINT=true
```
Comment (pyhalov) [inline components/network/bridge/Makefile]: Why do we need this?

File: components/network/tuntap/Makefile [hunk 6]
```diff
@@ -48,10 +51,6 @@ CONFIGURE_OPTIONS.64 +=  --enable-64bit
 PKGLINT=true
```
Comment (pyhalov) [inline components/network/tuntap/Makefile]: Is it still necessary?

**Comment by jimklimov** [comment]
No idea, artifact of copypaste. ;)

## 

Typos courtesy of GMail on my Samsung Android

22.11.2016 7:29 пользователь "Alexander Pyhalov" notifications@github.com
написал:

**Comment by jimklimov** [comment]
Revised and updated, thanks for the comments.

File: components/network/tuntap/tuntap-docs.p5m [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 Jim Klimov
+#
+
+set name=pkg.fmri value=pkg:/driver/network/tuntap-docs@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
```
Comment (pyhalov) [inline components/network/tuntap/tuntap-docs.p5m]: Please, rename it to driver/network/tuntap/documentation... It seems to be a more common pattern. Or perhaps even deliver it as documentation/tuntap, as it's not a driver

Comment (pyhalov) [reply components/network/tuntap/tuntap-docs.p5m]: Otherwise, whole PR seems fine

Comment (jimklimov) [reply components/network/tuntap/tuntap-docs.p5m]: done

**Pull request merged**